crates/macro-support/src/parser.rs

```diff
@@ -754,11 +785,21 @@ fn function_from_decl(
         })
     };
 
+    let replace_colliding_arg = |i: &mut syn::PatType| {
+        if let syn::Pat::Ident(ref mut i) = *i.pat {
+            let ident = i.ident.to_string();
+            if is_js_keyword(ident.as_str()) {
+                i.ident = Ident::new(format!("_{}", ident).as_str(), i.ident.span());
```

Comment:
This should not silently replace the name of exported and imported identifiers where they make a semantic difference. Accessing them by bracket instead of dot syntax sounds like a reasonable workaround. I.e. `wasm["class"]` instead of silently changing to `wasm._class` in

https://github.com/rustwasm/wasm-bindgen/pull/2855/files#diff-f7d340b325e0e9f30bd18d30036698d1edfece6974682dd3480457e4383806d3R22